ui/guest/src/store/epics/root.ts

```diff
@@ -669,15 +669,34 @@ const epic = combineEpics<GuestStandardAction, GuestStandardAction, GuestState>(
   },
   // endregion
   // region trashed
-  (action$: Observable<GuestStandardAction<{ iceId: number }>>) => {
+  (action$: Observable<GuestStandardAction<{ iceId: number }>>, state$) => {
     // onDrop doesn't execute when trashing on host side
     // Consider behaviour when running Host Guest-side
     return action$.pipe(
       ofType(trashed.type),
-      tap((action) => {
+      withLatestFrom(state$),
+      tap(([action, state]) => {
         const { iceId } = action.payload;
         let { modelId, fieldId, index } = iceRegistry.getById(iceId);
-        contentController.deleteItem(modelId, fieldId, index);
+
+        const { username, activeSite } = state;
+        const models = getCachedModels();
+        let parentModelId = getParentModelId(modelId, models, modelHierarchyMap);
+        let path = models[modelId].craftercms.path ?? models[parentModelId].craftercms.path;
+
+        if (isInheritedField(modelId, fieldId)) {
+          const modelIdToLock = getModelIdFromInheritedField(modelId, fieldId);
+          const parentModelId = getParentModelId(modelIdToLock, models, modelHierarchyMap);
+          path = models[parentModelId ?? modelIdToLock].craftercms.path;
+        }
+        beforeWrite$({
```

Comment:
This returns an observable, shouldn't we be switchMapping to the beforeWrite$ here?

Reply:
I can see the code. I'm asking we should rather be doing what I mentioned above.

Reply:
One thing that changes is that the computedDragEnd gets fired after beforeWrite$ actions, but it does get triggered properly